### PR TITLE
Disallow reserved words for object binding pattern names

### LINKED ISSUE #2722: Disallow reserved word in object binding pattern
Currently we allow the use of reserved word in object binding pattern:
```ts
var {while} = {while:1}
```
This is incorrect since while is a reserved word.

### PR DESCRIPTION
**Pull request by DanielRosenwasser**
Fixes #2722.

### CODE REVIEW

**Comment by mhegazy** [comment]
:+1:

File: src/compiler/parser.ts [hunk 1]
```diff
@@ -3859,13 +3859,14 @@ module ts {
         function parseObjectBindingElement(): BindingElement {
             let node = <BindingElement>createNode(SyntaxKind.BindingElement);
             // TODO(andersh): Handle computed properties
-            let id = parsePropertyName();
-            if (id.kind === SyntaxKind.Identifier && token !== SyntaxKind.ColonToken) {
-                node.name = <Identifier>id;
+            let tokenIsIdentifier = isIdentifier();
```
Comment (CyrusNajmabadi) [inline src/compiler/parser.ts]: Did we decide to not just always parse out a property name, and then error later if the property name turns out to be a keyword?

**Comment by JsonFreeman** [comment]
Looks like we get this wrong for import / export specifiers as well! I tried the following:
```ts
interface StringTree { }
export { StringTree as while };
```
I got no error.

File: src/compiler/parser.ts (hunk 1 above)
Comment (JsonFreeman) [inline src/compiler/parser.ts]: That is another approach we are considering. I do prefer it

**Pull request merged**